app/browser/api/ledger.js

```diff
@@ -0,0 +1,2187 @@
+  let keys = null
+  const fs = require('fs')
+  let data = fs.readFileSync(filePath)
+
+  if (!data || !data.length || !(data.toString())) {
+    state = logError(state, 'No data in backup file', 'recoveryWallet')
+  } else {
+    try {
+      const recoveryFileContents = data.toString()
+
+      let messageLines = recoveryFileContents.split(os.EOL)
+
+      let paymentIdLine = '' || messageLines[3]
+      let passphraseLine = '' || messageLines[4]
+
+      const paymentIdPattern = new RegExp([locale.translation('ledgerBackupText3'), '([^ ]+)'].join(' '))
+      const paymentId = (paymentIdLine.match(paymentIdPattern) || [])[1]
+
+      const passphrasePattern = new RegExp([locale.translation('ledgerBackupText4'), '(.+)$'].join(' '))
+      const passphrase = (passphraseLine.match(passphrasePattern) || [])[1]
+
+      keys = {
+        paymentId,
+        passphrase
+      }
+    } catch (exc) {
+      state = logError(state, exc, 'recoveryWallet')
+    }
+  }
+
+  return {
+    state,
+    keys
+  }
+}
+
+const getPublisherData = (result, scorekeeper) => {
+  let duration = result.duration
+
+  let data = {
+    verified: result.options.verified || false,
+    site: result.publisherKey,
+    views: result.visits,
+    duration: duration,
+    daysSpent: 0,
+    hoursSpent: 0,
+    minutesSpent: 0,
+    secondsSpent: 0,
+    faviconURL: result.faviconURL,
+    score: result.scores[scorekeeper],
+    pinPercentage: result.pinPercentage,
+    weight: result.pinPercentage
+  }
+  // HACK: Protocol is sometimes blank here, so default to http:// so we can
+  // still generate publisherURL.
+  data.publisherURL = (result.protocol || 'http:') + '//' + result.publisherKey
+
+  if (duration >= miliseconds.day) {
+    data.daysSpent = Math.max(Math.round(duration / miliseconds.day), 1)
+  } else if (duration >= miliseconds.hour) {
+    data.hoursSpent = Math.max(Math.floor(duration / miliseconds.hour), 1)
+    data.minutesSpent = Math.round((duration % miliseconds.hour) / miliseconds.minute)
+  } else if (duration >= miliseconds.minute) {
+    data.minutesSpent = Math.max(Math.round(duration / miliseconds.minute), 1)
+    data.secondsSpent = Math.round((duration % miliseconds.minute) / miliseconds.second)
+  } else {
+    data.secondsSpent = Math.max(Math.round(duration / miliseconds.second), 1)
+  }
+
+  return data
+}
+
+const normalizePinned = (dataPinned, total, target, setOne) => dataPinned.map((publisher) => {
+  let newPer
+  let floatNumber
+
+  if (setOne) {
+    newPer = 1
+    floatNumber = 1
+  } else {
+    floatNumber = (publisher.pinPercentage / total) * target
+    newPer = Math.floor(floatNumber)
+    if (newPer < 1) {
+      newPer = 1
+    }
+  }
+
+  publisher.weight = floatNumber
+  publisher.pinPercentage = newPer
+  return publisher
+})
+
+// courtesy of https://stackoverflow.com/questions/13483430/how-to-make-rounded-percentages-add-up-to-100#13485888
+const roundToTarget = (l, target, property) => {
```

Comment:
this might be good in ledgerUtil... although, for now it's fine to keep here since it's still using underscore.  Would be great to have a unit test for this though 😄

Reply:
yeah my plan for later is to remove underscore and move even more functions into util. Current ledger file is better, but still have a lot of function (which some you can't unit test) in single file